Fix getBuildPluginFile for examples and thirdparty build

breskeby opened the pull request:
This fixes our integration tests resolving roles.yaml for our examples build where the root is
not pointing to elasticsearch root folder

breskeby committed:
Fix getBuildPluginFile for examples build

This fixes our integration tests resolving roles.yaml for our examples build where the root is
not pointing to elasticsearch root folder

breskeby committed:
Fix typo

breskeby committed:
Add error handling for non anticipated build settings

breskeby committed:
Add some more debug information

breskeby committed:
Simplify buildpluginFile for non elasticsearch builds

breskeby committed:
Fix formatting

mark-vieira commented:
This is not the right fix IMO. We should not have explicit references to locations in our build in _any_ code in `build-tools` since that is publicly published. Instead this code should move into `build-tools-internal` and we add logic in `InternalTestClustersPlugin` to register that roles file.

@tvernum this is related to your changes in https://github.com/elastic/elasticsearch/pull/81400. I have a few questions:

1. Is that roles file _always_ required. That is, for external plugin authors, do they need that roles file to successfully setup a test clsuter, or are we simply implicitly relying on these defaults across our own tests.
2. I see that you PR added two of these files, once in `build-tools` and another in `build-tools-internal`, yet I see no references to the latter. Which one do we need and in which context?

elasticmachine commented:
Pinging @elastic/es-delivery (Team:Delivery)

breskeby commented:
total agree 100%. Somehow overlooked it's `build-tools`

breskeby committed:
Resolve default roles.yaml via TextResourceHandler

This fixes plugin usage for external builds and fixes relying on internal file structures

breskeby commented:
@mark-vieira I reworked the fix to rely on TextResourceHandler which also makes this work for external builds. It seems indeed the roles.yml file in `build-tools-internal` was accidentally added? @tvernum

breskeby committed:
Remove unused roles.yml file

breskeby committed:
Minor polishing

mark-vieira commented on build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java:
```diff
@@ -759,7 +764,8 @@ public void user(Map<String, String> userSpec) {
     private File getBuildPluginFile(String name) {
-        return project.getRootProject().file("build-tools/src/main/resources/" + name);
+        URL resource = getClass().getResource(name);
+        return fileOperations.getResources().getText().fromUri(resource).asFile();
```
Nice! I forgot we can use `TextResources` to load stuff off the classpath. We should probably replace [stuff like this](https://github.com/elastic/elasticsearch/blob/master/build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/info/GlobalBuildInfoPlugin.java#L358-L374) as well.

mark-vieira reviewed (approved):
LGTM. Thanks, Rene.

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [8.0](https://github.com/elastic/elasticsearch/pull/82815)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/82815">](https://github.com/elastic/elasticsearch/pull/82815) |
